### DISCUSSION TITLE
Tracked .slp file contains no points and nan for predictions

### DISCUSSION THREAD

**Discussion by scowlingwind** [talmolab/sleap#1225] [Help!] [answered]
Hello all. I have a trained model and I ran the following command to track a video -

`sleap-track 300ms.mp4 -m path/to/model/`

However afterwards when I open the file `300ms.mp4.predictions.slp` in `sleap-label` it does not show any points. opening the file in python and seeing the `"pred_points"` and `"points"` groups I don't get anything -

```ipython
In [1]: import h5py

In [2]: f = h5py.File("300ms.mp4.predictions.slp", "r")

In [3]: list(f.keys())
Out[3]: 
['frames',
 'instances',
 'metadata',
 'points',
 'pred_points',
 'suggestions_json',
 'tracks_json',
 'videos_json']

In [4]: list(f["points"])
Out[4]: []

In [5]: list(f["pred_points"])
Out[5]: 
[(nan, nan, True, False, 0.),
 (nan, nan, True, False, 0.),
 (nan, nan, True, False, 0.),
 (nan, nan, True, False, 0.),
 (nan, nan, True, False, 0.),
 (nan, nan, True, False, 0.),
...
```

Please let me know if I can provide anything else.

**Comment by roomrys** [comment]
Hi @sushmita-anant,

One possible cause could be that the you have [marked the nodes as occluded](https://sleap.ai/tutorials/initial-labeling.html?highlight=occluded#labeling-the-first-frame:~:text=For%20body%20parts%20that%20are%20not%20visible%20in%20the%20frame%2C%20right%2Dclick%20the%20node%20(or%20its%20name)%20to%20toggle%20visibility.%20The%20node%20will%20appear%20smaller%20to%20show%20that%20it%E2%80%99s%20marked%20as%20%E2%80%9Cnot%20visible%E2%80%9D.) in your initial labeling (i.e. if you marked some nodes as occluded in a previous instance and then created subsequent instances this occlusion would have carried over since SLEAP copy/pastes the last instance). Do you see many nodes that are smaller and have a gray highlight around the node name? This would mean that the node is marked as occluded. To toggle node visibility, right click on the node.

![image](https://user-images.githubusercontent.com/38435167/226356904-d90af50b-74ad-4682-aa45-273895d3313b.png)
**Image**: The "hindlegR4" node is occluded while all other node are marked as visible.

If the node are not unintentionally marked as occluded, then another possibility is that you model trained poorly which is a sign your parameters need tuning. To help diagnose this, could you include a screenshot of your model configuration(s) shown in the Training Pipeline GUI (please include the blue box overlaid on a frame representing the receptive field).

Let us know if this helps!

Thanks,
Liezl

**Reply by scowlingwind** [reply]
Hello @roomrys,

There haven't been any unintentionally occluded points in my labelled dataset. I'm tracking the motion of zebrafish larvae using the following hyperparameters: 
<img width="1381" alt="Screenshot 2023-03-22 at 00 31 56" src="https://user-images.githubusercontent.com/128314428/226714211-8a581f14-72fa-4af3-a323-52e7b36721a2.png">
Additionally, the training stops after a little over 20 epochs and there is an error in computing the metrics of the trained model. All the predicted instances have occluded labels.
Please let me know if I can provide anything else. Thanks so much!

**Reply by roomrys** [reply]
Hi @sushmita-anant,

Can you try increasing the input scaling (>0.5 but <=1, maybe 0.75) and using the max stride to keep the receptive field (blue box) the size of your largest zebra fish? It's possible we are throwing out valuable pixels especially if you are tracking the thin tail. However, I would still expect some predictions at least for the head... what does your skeleton look like? Do you have any success with either a top-down or bottom-up model?

Thanks,
Liezl

**Reply by scowlingwind** [reply, marked as answer]
Hi @roomrys,

I was able to generate predictions successfully using multi-animal topdown by making some changes to my skeleton and updating to the latest stable version of SLEAP on my local computer. Thank you so much for your suggestions and help and apologies for the delayed response.
